mantidproject/mantid
# Abins support for VASP calculations

**Description of work.**

Requested by ISIS molecular spectroscopy group.

VASP is a popular code in solid-state chemistry community, especially outside the UK. The inbuilt vibrations facilities are limited to Gamma-point calculations, and extended phonon modelling requires external tools such as Phonopy. Nonetheless, for many users an IBRION=8 calculation will be convenient and provide sufficient information, so we provide some support to directly read frequencies and eigenvectors from such calculations.

- [x] Implement OUTCAR reading
- [x] Implement vasprun.xml reading
- [x] Benchmark against CASTEP
- [x] _Consider removing/hiding OUTCAR support; the XML format should be more robust and maintainable against VASP revisions. However, the plain-text format is more familiar to users and helpful for initial benchmarking as e.g. typical working units are displayed._ **Decision made and implemented:** no mention of OUTCAR in main user docs, explain in concepts docs, print warning on use.

**To test:**

- [x] Add test data to repository

A unit test has been added for the LoadVASP module which may be executed with `ctest -R AbinsLoadVASP.

To test a user scenario, open the the Abins algorithm from simulations, set the ab initio code to VASP (which should be the bottom option in the drop-down list), unzip the following test file for benzene. Add the second-order quantum events if you like and set a workspace before running the algorithm.
[benzene-vasprun.zip](https://github.com/mantidproject/mantid/files/4479918/benzene-vasprun.zip)

To compare the results with CASTEP, here is the corresponding CASTEP unit test data checksum: https://github.com/mantidproject/mantid/blob/master/Testing/Data/UnitTest/benzene_Abins.phonon.md5
Bear in mind that these are not identical calculation setups, so exact agreement is not expected.

**Release notes**
- [x] Write release note

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
